### PR TITLE
Mrf 8868/add analytics post body

### PR DESCRIPTION
**Pull request by joselanderosmrf**

### CODE REVIEW

File: extensions/amp-analytics/0.1/transport.js [hunk 1]
```diff
@@ -82,12 +83,13 @@ export class Transport {
    * @param {!Window} win
    * @param {string} request
    * @return {boolean} True if this browser supports navigator.sendBeacon.
+   * @param {string} body
```
Comment (oliverfernandez) [inline extensions/amp-analytics/0.1/transport.js]: Mark this as optional?
`@param {string=} body`

File: extensions/amp-analytics/0.1/transport.js [hunk 2]
```diff
@@ -98,8 +100,9 @@ export class Transport {
    * @param {!Window} win
    * @param {string} request
    * @return {boolean} True if this browser supports cross-domain XHR.
+   * @param {string} body
```
Comment (oliverfernandez) [inline extensions/amp-analytics/0.1/transport.js]: Same optional here

File: extensions/amp-analytics/0.1/transport.js [hunk 3]
```diff
@@ -32,22 +32,23 @@ const TAG_ = 'amp-analytics.Transport';
  * @param {!Window} win
  * @param {string} request
  * @param {!Object<string, string>} transportOptions
+ * @param {string} body
  */
-export function sendRequest(win, request, transportOptions) {
+export function sendRequest(win, request, transportOptions, body) {
```
Comment (oliverfernandez) [inline extensions/amp-analytics/0.1/transport.js]: Who is calling this method? Is there no need to change the caller of this method to pass the body?

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 4]
```diff
@@ -898,12 +899,15 @@ export class AmpAnalytics extends AMP.BaseElement {
...
+      this.config_['transport']['iframe']) {
       user().assert(this.iframeTransport_,
           'iframe transport was inadvertently deleted');
       this.iframeTransport_.sendRequest(request);
     } else {
-      sendRequest(this.win, request, this.config_['transport'] || {});
+      sendRequest(this.win,
+          request,
+          this.config_['transport'] || {},
+          body || '');
```
Comment (joselanderosmrf) [inline extensions/amp-analytics/0.1/amp-analytics.js]: this is the caller

File: extensions/amp-analytics/0.1/transport.js [hunk 5]
```diff
@@ -32,22 +32,23 @@ const TAG_ = 'amp-analytics.Transport';
  * @param {!Window} win
  * @param {string} request
  * @param {!Object<string, string>} transportOptions
+ * @param {string} body
```
Comment (oliverfernandez) [inline extensions/amp-analytics/0.1/transport.js]: `@param {!string} body`

File: extensions/amp-analytics/0.1/requests.js [hunk 6]
```diff
@@ -239,13 +257,17 @@ export class RequestHandler {
...
-              this.constructBatchSegments_(baseUrl, batchSegmentsPromise);
+            this.constructBatchSegments_(baseUrl,
+                batchSegmentsPromise);
         } else {
           requestUrlPromise =
-              this.constructExtraUrlParamStrs_(baseUrl, extraUrlParamsPromise);
+            this.constructExtraUrlParamStrs_(baseUrl,
+                extraUrlParamsPromise);
         }
         requestUrlPromise.then(requestUrl => {
```
Comment (oliverfernandez) [inline extensions/amp-analytics/0.1/requests.js]:
````javascript
Promise.all([this.requestUrlPromise, this.requestBodyPromise]).then(...)

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 4 above)
Comment (oliverfernandez) [inline extensions/amp-analytics/0.1/amp-analytics.js]: body

**Pull request merged**